consensus/currency.go

```diff
@@ -0,0 +1,114 @@
+package consensus
+
+import (
+	"errors"
+	"math/big"
+)
+
+var (
+	ErrOverflow = errors.New("Currency overflowed 128 bits")
+)
+
+// A Currency is a 128-bit unsigned integer. Currency operations are performed
+// via math/big.
+//
+// The Currency object also keeps track of whether an overflow has occurred
+// during arithmetic operations. Once the 'overflow' flag has been set to
+// true, all subsequent operations will return an error, and the result of the
+// operation is undefined. This flag can never be reset; a new Currency must
+// be created. Callers can also manually check for overflow using the Overflow
+// method.
+type Currency struct {
+	i  big.Int
+	of bool // has an overflow ever occurred?
+}
+
+func NewCurrency(x uint64) Currency {
+	// no possibility of error
+	c, _ := BigToCurrency(new(big.Int).SetUint64(x))
+	return c
+}
```

Comment:
This might not be sufficient. uint64max is a lot, lot lower than the coin count per block (300,000 * 2^80). It works for testing though, which might be most of the purpose for this function.